browser/js/common/directives/navbar/navbar.js

```diff
@@ -1,47 +1,71 @@
-app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state) {
+app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state, $window) {
 
-    return {
-        restrict: 'E',
-        scope: {},
-        templateUrl: 'js/common/directives/navbar/navbar.html',
+	return {
+		restrict: 'E',
+		scope: {},
+		templateUrl: 'js/common/directives/navbar/navbar.html',
 
-        link: function (scope) {
+		link: function (scope) {
+			var temp = 0;
+			scope.goToSection = function(val, state) {
+				if (val !== 'null') {
+					Promise.resolve()
+					.then( () => {
+						if (!state) {
+							state = 'home';
+						}
+						if($state.current.name === state) {
+							return;
+						} else {
+						return $state.go(state);
+						}
+					})
+					.then( () => {
+						// temp = $window.pageYOffset;
+						// $window.pageYOffset -= 58;
+						smoothScroll(document.querySelector(val));
+					});
```

Comment:
I think if we want to do a `.catch` then we could just build a 404 page and have the user get redirected there. Or we could have a popup saying that page isn't available but it should never be the case that we have a link that doesn't redirect somewhere valid. And we can probably build that into our unit testing or something right.